src/stateMachines/OID4VCIMachine.tsx

```diff
@@ -0,0 +1,534 @@
+        invoke: {
+          src: OID4VCIMachineServices.retrieveCredentials,
+          onDone: {
+            target: OID4VCIMachineStates.verifyingCredentials,
+            actions: assign({credentialOffers: (_ctx: OID4VCIMachineContext, _event: DoneInvokeEvent<any>) => _event.data}),
+          },
+          onError: {
+            target: OID4VCIMachineStates.showingError,
+            actions: assign({
+              error: (_ctx: OID4VCIMachineContext, _event: DoneInvokeEvent<Error>): ErrorDetails => ({
+                title: 'retrieve credentials', // TODO translate
+                message: _event.data.message,
+              }),
+            }),
+          },
+        },
+        exit: assign({verificationCode: undefined}),
+      },
+      [OID4VCIMachineStates.verifyingCredentials]: {
+        id: OID4VCIMachineStates.verifyingCredentials,
+        invoke: {
+          src: OID4VCIMachineServices.verifyCredentials,
+          onDone: {
+            target: OID4VCIMachineStates.transitioningFromWalletInput,
+          },
+          onError: {
+            target: OID4VCIMachineStates.showingError,
+            actions: assign({
+              error: (_ctx: OID4VCIMachineContext, _event: DoneInvokeEvent<Error>): ErrorDetails => ({
+                title: 'verify credentials', // TODO translate
+                message: _event.data.message,
+              }),
+            }),
+          },
+        },
+      },
+      [OID4VCIMachineStates.transitioningFromWalletInput]: {
+        id: OID4VCIMachineStates.transitioningFromWalletInput,
+        always: [
+          {
+            target: OID4VCIMachineStates.addingContactIdentity,
+            cond: OID4VCIMachineGuards.contactIdentityGuard,
+          },
+          {
+            target: OID4VCIMachineStates.reviewingCredentialOffers,
+          },
+        ],
+      },
+      [OID4VCIMachineStates.addingContactIdentity]: {
+        id: OID4VCIMachineStates.addingContactIdentity,
+        invoke: {
+          src: OID4VCIMachineServices.addContactIdentity,
+          onDone: {
+            target: OID4VCIMachineStates.reviewingCredentialOffers,
+          },
+          onError: {
+            target: OID4VCIMachineStates.showingError,
+            actions: assign({
+              error: (_ctx: OID4VCIMachineContext, _event: DoneInvokeEvent<Error>): ErrorDetails => ({
+                title: 'adding contact identity', // TODO translate
+                message: _event.data.message,
+              }),
+            }),
+          },
+        },
+      },
+      [OID4VCIMachineStates.reviewingCredentialOffers]: {
+        id: OID4VCIMachineStates.reviewingCredentialOffers,
+        on: {
+          [OID4VCIMachineEvents.NEXT]: {
+            target: OID4VCIMachineStates.storingCredentialBranding,
+          },
+          [OID4VCIMachineEvents.DECLINE]: {
+            target: OID4VCIMachineStates.declined,
+          },
+          [OID4VCIMachineEvents.PREVIOUS]: {
```

Comment:
Should we allow a previous on this one. We are already at a NEXT or DECLINE state here. There is no going back, hance why you used the aborted state. If we would allow for a back button on a UI that would be a mistake IMO

Reply:
The header has a back button in the design, so we would have to fix it there. but android has a native back button as well. so there is always a back button option available.